chalicelib/checks/wrangler_checks.py

```diff
@@ -1887,3 +1887,110 @@ def check_external_references_uri(connection, **kwargs):
     check.brief_output = name_counts
     check.full_output = items
     return check
+
+
+@check_function(days_back=2)
+def check_opf_lab_different_than_experiment(connection, **kwargs):
+    '''
+    Check if other processed files have lab (generating lab) that is different
+    than the lab of that generated the experiment. In this case, the
+    experimental lab needs to be added to the opf (contributing lab).
+    '''
+    check = CheckResult(connection, 'check_opf_lab_different_than_experiment')
+
+    # check only recently modified files, to reduce the number of items
+    days_back = kwargs.get('days_back')
+    try:
+        days_back = float(days_back)
+    except (ValueError, TypeError):
+        from_date_query = ''
+        from_text = ''
+    else:
+        date_now = datetime.datetime.now(datetime.timezone.utc)
+        date_diff = datetime.timedelta(days=days_back)
+        from_date = datetime.datetime.strftime(date_now - date_diff, "%Y-%m-%d")
+        from_date_query = '&last_modified.date_modified.from=' + from_date
+        from_text = 'modified from %s ' % from_date
+
+    search = ('search/?type=FileProcessed' +
+              '&track_and_facet_info.experiment_bucket%21=No+value' +
+              '&track_and_facet_info.experiment_bucket%21=processed+file' +
+              '&field=experiment_sets&field=experiments' +
+              '&field=lab&field=contributing_labs' + from_date_query)
+    result = ff_utils.search_metadata(search, key=connection.ff_keys)
+
+    exp_set_uuids = []  # Exp or ExpSet uuid list
+    for res in result:
+        if res.get('experiments'):
+            if len(res['experiments']) != 1:
+                return check  # this should not happen
```

Comment:
is returning the check really the desired behavior here rather than flagging them in the output and returning all the proper results? If the check gets returned as this stage will it be obvious what the problem is?

Reply:
Good point. This is one of the parts that needed refinement. I initially wrote it this way based on the track_and_facet_info code, which does a similar thing. 
At the moment, no such file is found. I agree that the check should flag these cases and return all the proper results. However, I am not sure what to do next with these cases. If they will be manually patched, then probably they should also be added to a list of files to ignore?

Reply:
in the output you could separate actionable results and flagged results, and write the action so it only uses the actionable results. There probably doesn't need to be an "ignore" list, unless it is possible that some files won't meet that qualifier by design.